test/jasmine/tests/table_test.js

```diff
@@ -0,0 +1,972 @@
+                {values: [562, 124, 942]},
+                {values: [], visible: true},
+                {values: [1, 2], visible: false} // shouldn't be truncated to as false
+            ]});
+            expect(fullTrace.dimensions).toEqual([
+                {values: [], visible: true, valueformat: '3s', _index: 0},
+                {values: [], visible: true, valueformat: '3s', _index: 1},
+                {values: [], visible: true, valueformat: '3s', _index: 2},
+                {values: [1, 2], visible: false, _index: 3}
+            ]);
+        });
+    });
+
+    describe('table calc', function() {
+
+        function _calc(trace) {
+            var gd = { data: [trace] };
+
+            Plots.supplyDefaults(gd);
+
+            var fullTrace = gd._fullData[0];
+            Table.calc(gd, fullTrace);
+            return fullTrace;
+        }
+
+        var base = { type: 'table' };
+
+        it('\'colorscale\' should assume a default value if the \'color\' array is specified', function() {
+
+            var fullTrace = _calc(Lib.extendDeep({}, base, {
+                line: {
+                    color: [35, 63, 21, 42]
+                },
+                dimensions: [
+                    {values: [321, 534, 542, 674]},
+                    {values: [562, 124, 942, 189]},
+                    {values: [287, 183, 385, 884]},
+                    {values: [113, 489, 731, 454]}
+                ]
+            }));
+
+            expect(fullTrace.line).toEqual({
+                color: [35, 63, 21, 42],
+                colorscale: attributes.line.colorscale.dflt,
+                cauto: true,
+                cmin: 21,
+                cmax: 63,
+                autocolorscale: false,
+                reversescale: false,
+                showscale: false
+            });
+        });
+
+        it('use a singular \'color\' if it is not an array', function() {
+
+            var fullTrace = _calc(Lib.extendDeep({}, base, {
+                line: {
+                    color: '#444'
+                },
+                dimensions: [
+                    {values: [321, 534, 542, 674]},
+                    {values: [562, 124, 942, 189]}
+                ]
+            }));
+
+            expect(fullTrace.line).toEqual({
+                color: '#444'
+            });
+        });
+
+        it('use a singular \'color\' even if a \'colorscale\' is supplied as \'color\' is not an array', function() {
+
+            var fullTrace = _calc(Lib.extendDeep({}, base, {
+                line: {
+                    color: '#444',
+                    colorscale: 'Jet'
+                },
+                dimensions: [
+                    {values: [321, 534, 542, 674]},
+                    {values: [562, 124, 942, 189]}
+                ]
+            }));
+
+            expect(fullTrace.line).toEqual({
+                color: '#444'
+            });
+        });
+    });
+});
+
+describe('@noCI table', function() {
```

Comment:
Why `@noCI`?

Reply:
Basically I didn't intend to check in `table_test.js` 😃